lib/node_modules/@stdlib/lapack/base/dgebak/lib/dgebak.js

```diff
@@ -0,0 +1,98 @@
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var isLayout = require( '@stdlib/blas/base/assert/is-layout' );
+var isColumnMajor = require( '@stdlib/ndarray/base/assert/is-column-major-string' );
+var max = require( '@stdlib/math/base/special/max' );
+var isRowMajor = require( '@stdlib/ndarray/base/assert/is-row-major-string' );
+var format = require( '@stdlib/string/format' );
+var base = require( './base.js' );
+
+
+// MAIN //
+
+/**
+* Forms the right or left eigenvectors of a real general matrix by backward transformation on the computed eigenvectors of the balanced matrix output by `dgebal`.
+*
+* ## Notes
+*
+* The job parameter can be one of the following:
+*
+* -   `none`: none, return immediately
+* -   `permute`: permute only
+* -   `scale`: scale only
+* -   `both`: both permute and scale
+*
+* @param {string} order - storage layout
+* @param {string} job - type of transformation.
+* @param {string} side - specifies the side of the eigenvectors in `V` (`right` or `left`)
+* @param {NonNegativeInteger} N - number of rows in matrix `V`
+* @param {NonNegativeInteger} M - number of columns in matrix `V`
+* @param {NonNegativeInteger} ilo - index of leftmost column of the submatrix computed by `dgebal` (zero-based)
+* @param {NonNegativeInteger} ihi - index of rightmost column of the submatrix computed by `dgebal` (zero-based)
+* @param {Float64Array} scale - contains the details of the permutations and scaling factors applied when balancing `V`, computed by `dgebal`
+* @param {Float64Array} V - input matrix
+* @param {integer} LDV - leading dimension of `V`
+* @throws {TypeError} first argument must be a valid order
+* @throws {TypeError} second argument must be a valid job
+* @throws {TypeError} third argument must be a valid side
+* @throws {RangeError} fourth argument must be greater than or equal to max(1,M)
+* @returns {Float64Array} `V`
+*
+* @example
+* var Float64Array = require( '@stdlib/array/float64' );
+*
+* var scale = new Float64Array( [ 0.5, 2.0, 1.0 ] );
+* var V = new Float64Array( [ 1.0, 4.0, 2.0, 5.0, 3.0, 6.0 ] );
+*
+* var out = dgebak( 'row-major', 'scale', 'right', 3, 2, 0, 2, scale, V, 2 );
+* // returns <Float64Array>[ 0.5, 2, 4, 10, 3, 6 ]
+*/
+function dgebak( order, job, side, N, M, ilo, ihi, scale, V, LDV ) {
+	var sa1;
+	var sa2;
+	if ( !isLayout( order ) ) {
+		throw new TypeError( format( 'invalid argument. First argument must be a valid order. Value: `%s`.', order ) );
+	}
+	if ( job !== 'none' && job !== 'scale' && job !== 'permute' && job !== 'both' ) {
+		throw new TypeError( format( 'invalid argument. Second argument must be a valid job. Value: `%s`.', job ) );
+	}
+	if ( side !== 'left' && side !== 'right' ) {
```

Comment:
See if https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/blas/base/assert/is-operation-side is appropriate here.